packages/xo-web/src/common/bulk-icons.js

```diff
@@ -0,0 +1,63 @@
+import _ from 'intl'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import BaseComponent from './base-component'
+import Icon from './icon'
+import { alert } from './modal'
+import { createSelector } from './selectors'
+
+const ICON_WARNING_MODAL_LEVEL = {
+  danger: 1,
+  warning: 2,
+  success: 3,
+}
+
+class BulkIcons extends BaseComponent {
+  getSortedAlerts = createSelector(
+    () => this.props.alerts,
+    alerts =>
+      alerts?.sort((curr, next) => ICON_WARNING_MODAL_LEVEL[curr.level] - ICON_WARNING_MODAL_LEVEL[next.level]) ?? []
+  )
+
+  onClick = () =>
+    alert(
+      _('alerts'),
+      this.getSortedAlerts().map(({ level, render }, index) => (
+        <div className={`text-${level}`} key={index}>
+          {render}
+        </div>
+      ))
+    )
+
+  render() {
+    const alerts = this.getSortedAlerts()
+    const length = alerts.length
+    const level = alerts[0]?.level
+
+    return (
+      length !== 0 && (
+        // <a> in order to bypass the BlockLink component
+        <a className='fa-stack' onClick={this.onClick}>
+          <Icon icon='alarm' color={`text-${level}`} className='fa-stack-2x' />
+          {/* `fa-triangle` does not exist on FontAwesome4.`l` is used to fill the `!` of the `alarm` icon */}
```

Comment:
Is there a source for this `l is used to fill the ! of the alarm icon`? if so, add its link.

Reply:
There is no source for this hack.

Reply:
Shouldn't the `!` be squeezed by the number?